master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/AbstractMetaManager.java

```diff
@@ -64,7 +66,7 @@ public abstract class AbstractMetaManager implements IMetadataHandler {
   public final Map<String, Set<Integer>> registeredAppAndShuffles =
       JavaUtils.newConcurrentHashMap();
   public final Set<String> hostnameSet = ConcurrentHashMap.newKeySet();
-  public final Set<WorkerInfo> workers = ConcurrentHashMap.newKeySet();
+  private final Map<Integer, WorkerInfo> workersMap = JavaUtils.newConcurrentHashMap();
```

Comment:
I think we can extract a workerinfo pool in Celeborn master and use  `Set<String>` to track workerinfo objects.
Workerinfo object has a method to get unique identifier called `toUniqueId`.

Reply:
If you choose the hash code as the key, you may encounter hash collisions. @turboFei

Reply:
thanks for the suggestion, got it